scripts/gen-webhook-secret.sh

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
+
+# Use this script to update the webhook secret post-deployment. Alternatively,
+# you can use patch-deploy-yaml.sh prior to deployment.
+#
+# Generates a certificate for use by the controller webhook
+# You can also manually re-create the webhook secret using your own PKI
+
+TEMP_KEY=`mktemp`
+TEMP_CERT=`mktemp`
+
+WEBHOOK_SVC_NAME=webhook-service
+WEBHOOK_NAME=aws-appnet-gwc-mutating-webhook
+WEBHOOK_NAMESPACE=aws-application-networking-system
+WEBHOOK_SECRET_NAME=webhook-cert
+
+echo "Generating certificate for webhook"
+HOST=${WEBHOOK_SVC_NAME}.${WEBHOOK_NAMESPACE}.svc
+openssl req -x509 -nodes -days 36500 -newkey rsa:2048 -keyout ${TEMP_KEY} -out ${TEMP_CERT} -subj "/CN=${HOST}/O=${HOST}" \
+   -addext "subjectAltName = DNS:${HOST}, DNS:${HOST}.cluster.local"
+
+export KEY_B64=`cat $TEMP_KEY | base64`
+export CERT_B64=`cat $TEMP_CERT | base64`
+
+echo "Recreating webhook secret"
+kubectl delete secret $WEBHOOK_SECRET_NAME --namespace $WEBHOOK_NAMESPACE
+kubectl create secret tls $WEBHOOK_SECRET_NAME --namespace $WEBHOOK_NAMESPACE --cert=${TEMP_CERT} --key=${TEMP_KEY}
+
+echo "Patching webhook with new cert"
+kubectl patch mutatingwebhookconfigurations.admissionregistration.k8s.io $WEBHOOK_NAME \
+    --namespace $WEBHOOK_NAMESPACE --type='json' \
+    -p="[{'op': 'replace', 'path': '/webhooks/0/clientConfig/caBundle', 'value': '${CERT_B64}'}]"
+
+rm $TEMP_KEY $TEMP_CERT
+echo "Done"
```

Comment:
Example output:
```
scripts/gen-webhook-secret.sh 
Generating certificate for webhook
..+++++++++++++++++++++++++++++++++++++++*.+...............+.+......+...+..+.+............+++++++++++++++++++++++++++++++++++++++*...+......+................+........+......+.+.....+.+...........................+.....+......+....+...+..............+...+......+.+......+.........+.....+.+..................+..+...+....+......+......+...+.....+...+....+...+......+....................+.+.....+......+.......+..++++++
...........+.+.....+.........+......+.......+...............+..+......+....+........+++++++++++++++++++++++++++++++++++++++*....+...+.+......+..............+.+...........+.......+.....+.......+.....+...+.+..+...+.+.....+.+.....+....+......+...+.....+++++++++++++++++++++++++++++++++++++++*...+.+.....+.........+.+......+...+......................................+...+.+......+...............+.....+.........+....+..+.............+..+...+.+.....+.+...+....................+......+...+......+.+.....+......+.......+...+..+...+..................+.+...+.....+.+.........+..+...+.............+..+...............+.............+...+...........+...+..........++++++
-----
Recreating webhook secret
secret "webhook-cert" deleted
secret/webhook-cert created
Patching webhook with new cert
mutatingwebhookconfiguration.admissionregistration.k8s.io/aws-appnet-gwc-mutating-webhook patched
Done
```

Reply:
Double confirm, doing a `kubectl delete secret webhook-cert`  for a running pod with
```
      volumes:
        - name: webhook-cert
          secret:
            defaultMode: 420
            secretName: webhook-cert
```
is all fine, right?

Do we need to print log here, say, the user should reboot the controller to get the new `webhook-cert` content and set WEBHOOK_ENABLED==true to make webhook work?

Reply:
> Just double confirm, does gen-webhook-secret.sh work for both deploy.yaml deployed and helm installed controller?

Correct.

> kubectl delete secret webhook-cert for a running pod

I have recreated the secret before for a running pod. It does work. I haven't tested what happens if you only delete but don't recreate.